crates/mcp-client/src/transport/mod.rs

```diff
@@ -0,0 +1,79 @@
+use async_trait::async_trait;
+use std::sync::Arc;
+use thiserror::Error;
+use tokio::sync::Mutex;
+
+/// A generic error type for transport operations.
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("I/O error: {0}")]
+    Io(#[from] std::io::Error),
+
+    #[error("Transport was not connected or is already closed")]
+    NotConnected,
+
+    #[error("Invalid URL provided")]
+    InvalidUrl,
+
+    #[error("Connection timeout")]
+    Timeout,
+
+    #[error("Failed to send message")]
+    SendFailed,
+
+    #[error("Unexpected transport error: {0}")]
+    Other(String),
+}
+
+/// A generic asynchronous transport trait.
+///
+/// Implementations are expected to handle:
+/// - starting the underlying communication channel (e.g., launching a child process, connecting a socket)
+/// - sending JSON-RPC messages as strings
+/// - receiving JSON-RPC messages as strings
+/// - closing the transport cleanly
+#[async_trait]
+pub trait Transport: Send + 'static {
```

Comment:
start/close make sense, but i think we want to have a Receiver and Sender so we can more carefully handle parallelism. there needs to be something polling these to route responses back to their originating requests, and implementing send/receive from scratch would mean we need to be more careful with that handling - i wouldn't want to have to reimplement the sleep vs close etc logic